src/darwin/Framework/CHIPTests/MTRPerControllerStorageTests.m

```diff
@@ -2298,4 +2298,496 @@ - (void)testSubscriptionPool
+    //    That should cause the endpoint and its corresponding clusters to be removed from data storage.
+    // 2. The data store is populated with cluster index and cluster data for endpoints 0, 1 and 2 initially.
+    // 3. After the fake attribute report is injected with deleted endpoint 2, make sure the data store is still populated with cluster index and cluster data
+    // for endpoints 0 and 1 but not 2.
+    __block NSDictionary * testDataForPartsList;
+    __block NSMutableArray * testClusterDataValueForPartsList;
+    delegate.onAttributeDataReceived = ^(NSArray<NSDictionary<NSString *, id> *> * attributeReport) {
+        attributeReportsReceived += attributeReport.count;
+        XCTAssert(attributeReportsReceived > 0);
+
+        for (NSDictionary<NSString *, id> * attributeDict in attributeReport) {
+            MTRAttributePath * attributePath = attributeDict[MTRAttributePathKey];
+            XCTAssertNotNil(attributePath);
+
+            if ([attributePath.endpoint isEqualToNumber:rootEndpoint] && attributePath.cluster.unsignedLongValue == MTRClusterIDTypeDescriptorID && attributePath.attribute.unsignedLongValue == MTRAttributeIDTypeClusterDescriptorAttributePartsListID) {
+                testDataForPartsList = attributeDict[MTRDataKey];
+                XCTAssertNotNil(testDataForPartsList);
+                dataVersionForPartsList = testDataForPartsList[MTRDataVersionKey];
+                id dataValue = testDataForPartsList[MTRValueKey];
+                XCTAssertNotNil(dataValue);
+                testClusterDataValueForPartsList = [dataValue mutableCopy];
+            }
+        }
+    };
+
+    __block NSMutableDictionary<NSNumber *, NSArray<NSNumber *> *> * initialClusterIndex = [[NSMutableDictionary alloc] init];
+    __block NSMutableArray * testEndpoints;
+
+    delegate.onReportEnd = ^{
+        XCTAssertNotNil(dataVersionForPartsList);
+        XCTAssertNotNil(testClusterDataValueForPartsList);
+        testEndpoints = [self getEndpointArrayFromPartsList:testDataForPartsList forDevice:device];
+
+        // Make sure that the cluster data in the data storage is populated with cluster index and cluster data for endpoints 0, 1 and 2.
+        // We do not need to check _persistedClusterData here. _persistedClusterData will be paged in from storage when needed so
+        // just checking data storage should suffice here.
+        dispatch_sync(self->_storageQueue, ^{
+            XCTAssertTrue([[controller.controllerDataStore _fetchEndpointIndexForNodeID:deviceID] isEqualToArray:testEndpoints]);
+
+            // Populate the initialClusterIndex to use as a reference for all cluster paths later.
+            for (NSNumber * endpoint in testEndpoints) {
+                [initialClusterIndex setObject:[controller.controllerDataStore _fetchClusterIndexForNodeID:deviceID endpointID:endpoint] forKey:endpoint];
+            }
+            XCTAssertNotNil(initialClusterIndex);
```

Comment:
That's a strange thing to be asserting given we were just calling things on it....

Reply:
this is not needed.